File: FactoryExample/src/factoryexample/ArticuloOficina.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package factoryexample;

/**
 *
 * @author fabm_
 */
public class ArticuloOficina extends Articulo{
    
    public ArticuloOficina(int id, String nombre, TipoArticulo tipoArticulo, long precio) {
        super(id, nombre, tipoArticulo, precio);
    }

    @Override
    public long calcularIva() {
        if(this.getPrecio() == 0){
            return 0;
        }else{
            long iva = (long) (0.21 * this.getPrecio());
            return iva;
        }
        
    }
    
}

File: FactoryTs/articulooficina.ts
import { Articulo } from "./articulo";
import { TipoArticulo } from "./tipoarticulo";

export class ArticuloOficina extends Articulo{

    constructor(idArticulo:number, 
                nombre:string, 
                tipoArticulo:TipoArticulo, 
                precio:number){
            
                    super(idArticulo,nombre,tipoArticulo,precio);
    }

    public calcularIva():number {
        if(this.getPrecio() == 0){
            return 0;
        }else{
            console.log(0.21*this.getPrecio());
            return 0.21 * this.getPrecio();
        }
        
    }
}